sips/sip-tbd/sip-TBD-token-metadata-update-notifications.md

```diff
@@ -0,0 +1,255 @@
+  updates.
+  * Refreshing a token's metadata should be an idempotent operation. Repeated refreshes should not
+    create distinct records in the internal metadata database.
+* Notifications could be gratuitous.
+  * To prevent slow performance and guard against any Denial of Service attack attempts, contract
+    call rate limiting should be implemented locally.
+* Notifications can be delayed and out of order.
+  * A notification transaction's timestamp should not be considered to be the time when the token
+    metadata was actually updated.
+
+# Backwards compatibility
+
+Developers who need to emit metadata update notifications for tokens declared in older contracts
+(that were deployed before this notification standard was established) could do so by either calling
+the contract described in [Reference Implementations](#reference-implementations) or by deploying a
+new separate contract containing a public function that emits this notification.
+
+# Activation
+
+This SIP will be activated when at least 10 unique contracts have had metadata updates triggered via
+contract-call transactions that print the proposed notification payload. If the Stacks blockchain
+reaches block height 170000 and this has not happened, this SIP will be considered rejected.
+
+# Reference implementations
+
+A [reference contract](./token-metadata-update-notify.clar) has been deployed to mainnet as
+[`SP1H6HY2ZPSFPZF6HBNADAYKQ2FJN75GHVV95YZQ.token-metadata-update-notify`](https://explorer.stacks.co/txid/0xe92af2ea5c11e2e6fde4d31fd394de888070efff23bffad04465c549543abfa2?chain=mainnet).
+It demonstrates how to send notifications for each token class and it is available for developers to
+use for refreshing any existing or future token contract. If the SIP evolves to require a change to
+this contract pre-activation, a new one will be deployed and noted here.
+
+```clarity
+;; token-metadata-update-notify
+;;
+;; Use this contract to notify token metadata indexers that an NFT or FT needs its metadata
+;; refreshed.
+
+(use-trait nft-trait 'SP2PABAF9FTAJYNFZH93XENAJ8FVY99RRM50D2JG9.nft-trait.nft-trait)
+(use-trait ft-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
+
+;; Refresh the metadata for one or more NFTs from a specific contract.
+(define-public (nft-metadata-update-notify (contract <nft-trait>) (token-ids (list 100 uint)))
+  (ok (print
+    {
+      notification: "token-metadata-update",
+      payload: {
+        contract-id: contract,
+        token-class: "nft",
+        token-ids: token-ids
+      }
+    })))
+
+;; Refresh the metadata for a FT from a specific contract
+(define-public (ft-metadata-update-notify (contract <ft-trait>))
+  (ok (print
+    {
+      notification: "token-metadata-update",
+      payload: {
+        contract-id: contract,
+        token-class: "ft"
+      }
+    })))
+```
+
+The [Stacks Blockchain API](https://github.com/hirosystems/stacks-blockchain-api) will also add
```

Comment:
Should this be an activation criterion as well?

Reply:
Agreed, added a minimum of 3 indexers to the criteria.